opm/parser/eclipse/Parser/Parser.cpp

```diff
@@ -229,6 +229,7 @@ namespace Opm {
         parserState->openRootFile( dataFileName );
         parseState(parserState);
         applyUnitsToDeck(*parserState->deck);
+        parserState->deck->setDataFile(dataFileName);
```

Comment:
Here I would prefer:

1. The `parserState` instance internalizes the rootFilename.
2. The `Deck` constructor gets a `const std::string& inputFile`input argument.

Then we can manage without the `Deck::setDataFile( )` method.